### PR TITLE
#167750039 Setup Sequelize ORM with PostgreSQL

### PR DESCRIPTION
**Pull request by uthdev**
#### Description of Task to be completed?
- Setup Sequelize ORM with PostgreSQL
- Create models and migrations structure
- Create a Dummy User table
 #### How should this be manually tested?
- clone the repo
- cd into the root folder of the cloned repo
- run `npm install`  to install necessary dependencies
- run `npm install -g sequelize-cli` 
- open the terminal and type `touch .env` in the terminal
- ensure you have Postgres installed and setup
- create your database type  `createdb <db-name>`
- open the .env file and put your database info inside, check out the .env-example file in the root of the repo to see how to do this
- run `npm run migrate` to create your tables
 #### Any background context you want to provide?
 #### What are the relevant pivotal tracker stories?
[#167750044](https://www.pivotaltracker.com/story/show/167750044), [#167750039](https://www.pivotaltracker.com/story/show/167750039)
 #### Screenshots (if appropriate)
None
 #### Questions:
None

### CODE REVIEW

**Review by feobaby — COMMENTED** [review]
Good job Gbolahan, but we can see no link to the PT story?

**Comment by uthdev** [comment]
> Good job Gbolahan, but we can see no link to the PT story?

@funmi5, link how? Do you mean the Finishes hook at the end of the commit?

**Comment by feobaby** [comment]
> > Good job Gbolahan, but we can see no link to the PT story?
> 
> @funmi5, link how? Do you mean the Finishes hook at the end of the commit?

what i mean is the link /links to your pt stories, something like this `<a href = "<pivotal-tracker-link>">#33333</a>`

**Comment by uthdev** [comment]
> > > Good job Gbolahan, but we can see no link to the PT story?
> > 
> > 
> > @funmi5, link how? Do you mean the Finishes hook at the end of the commit?
> 
> what i mean is the link /links to your pt stories, something like this `<a href = "<pivotal-tracker-link>">#33333</a>`

I don't understand what you mean. Can you direct me on how I'm supposed to do that

**Comment by torsami77** [comment]
Good example of documented PR @uthdev

**Comment by sa-ma** [comment]
Good job @uthdev Some things to note

1. You have multiple commits
2. Your branch isn't updated with develop, you still have all those libraries I was told to uninstall in your package.json
3. The branch name is also not in line with the wiki

**Comment by uthdev** [comment]
> Good job @uthdev Some things to note
> 
> 1. You have multiple commits
> 2. Your branch isn't updated with develop, you still have all those libraries I was told to uninstall in your package.json
> 3. The branch name is also not in line with the wiki

Well, I didn't know which libraries to keep as I didn't know the ones he told you to remove. So while resolving the conflict when I was rebasing, I just accepted both cause I didn't know what to remove. I'll see if I can rebase again and push

**Comment by uthdev** [comment]
> > > Good job Gbolahan, but we can see no link to the PT story?
> > 
> > 
> > @funmi5, link how? Do you mean the Finishes hook at the end of the commit?
> 
> what i mean is the link /links to your pt stories, something like this `<a href = "<pivotal-tracker-link>">#33333</a>`

@funmi5 I've done that. You can check now. By the way, this didn't work cause it is not an html file. I used Markdown format

**Review by feobaby — COMMENTED** [review]
oh Good, but include the hashtag too.

**Comment by uthdev** [comment]
> oh Good, but include the hashtag too.

Done @funmi5

**Review by oyewoas — DISMISSED** [review]
Nice....

**Review by feobaby — COMMENTED** [review]
@uthdev Please, squash your commits into just one.

File: .env-example [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DATABASE_URL=
```
Comment (Oliver-ke) [inline .env-example:1]: hi @uthdev could you please check this line, I can't find its use

File: .env-example [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+DEV_DATABASE_URL=postgres://<db_user>:<db_password>@127.0.0.1:5432/dev_db
+TEST_DATABASE_URL=postgres://<db_user>:<db_password>@127.0.0.1:5432/test_db
+
+db_user: name of the owner of the database. Moslty always 'postgres'
+db_password: passwoed set for the database
```
Comment (Oliver-ke) [inline .env-example]: hi @uthdev , I see you connecting to the database using the URL, could there be any need for the `db_user` and `db_password` keys, Also they're not assigned to any specific information

**Review by Oliver-ke — COMMENTED** [review]
Hi @uthdev you've don great, however, you have seven commits, I suggest you squash them to get one, click [here](https://www.internalpointers.com/post/squash-commits-into-one-git) for a guide, finally following your pull description I suggest you'll be more specific as to what is the name of the database to create and how many databases

File: .env-example (hunk 1 above)
Comment (uthdev) [reply .env-example:1]: Thank you @Oliver-ke, that's that is just for the production database URL. Put it there so we know you'll also need in the environment variables. We are going to be having 3 databases each for development, testing and production. Check out the config file in `/src/database/config/` directory to confirm this.

File: .env-example (hunk 2 above)
Comment (uthdev) [reply .env-example]: @Oliver-ke this is just a guide on how to make the connection string. I put those there as additional information for anyone setting up to know what the db_user and db_password are. Not that they should create two new keys db_user and db_password after creating the connection string.

**Comment by sa-ma** [comment]
@uthdev you have merge conflicts on your PR

**Review by feobaby — DISMISSED** [review]
Finally, your sequelize set up is solid, I like the way you hid the core info of the dbs in the .env file, you squashed your commits into one and you resolved your conflicts. Great Job so far @uthdev

**Review by sa-ma — APPROVED** [review]
Everything looks ok, you'll have to refactor to change from es5 to es6

**Review by feobaby — APPROVED** [review]
yea, i think we have *only* the dependencies we need now.

**Pull request merged**